pkg/pxc/backup/restore.go

```diff
@@ -260,14 +260,117 @@ func S3RestoreJob(cr *api.PerconaXtraDBClusterRestore, bcp *api.PerconaXtraDBClu
 		jobPVC,
 		app.GetSecretVolumes("vault-keyring-secret", cluster.PXC.VaultSecretName, true),
 	}
+	pxcUser := "xtrabackup"
+	command := []string{"recovery-s3.sh"}
+
+	envs := []corev1.EnvVar{
+		{
+			Name:  "S3_BUCKET_URL",
+			Value: s3dest,
+		},
+		{
+			Name:  "ENDPOINT",
+			Value: bcp.Status.S3.EndpointURL,
+		},
+		{
+			Name:  "DEFAULT_REGION",
+			Value: bcp.Status.S3.Region,
+		},
+		{
+			Name: "ACCESS_KEY_ID",
+			ValueFrom: &corev1.EnvVarSource{
+				SecretKeyRef: &corev1.SecretKeySelector{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: bcp.Status.S3.CredentialsSecret,
+					},
+					Key: "AWS_ACCESS_KEY_ID",
+				},
+			},
+		},
+		{
+			Name: "SECRET_ACCESS_KEY",
+			ValueFrom: &corev1.EnvVarSource{
+				SecretKeyRef: &corev1.SecretKeySelector{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: bcp.Status.S3.CredentialsSecret,
+					},
+					Key: "AWS_SECRET_ACCESS_KEY",
+				},
+			},
+		},
+		{
+			Name:  "PXC_SERVICE",
+			Value: cr.Spec.PXCCluster + "-pxc",
+		},
+		{
+			Name:  "PXC_USER",
+			Value: pxcUser,
+		},
+		{
+			Name: "PXC_PASS",
+			ValueFrom: &corev1.EnvVarSource{
+				SecretKeyRef: app.SecretKeySelector(cluster.SecretsName, pxcUser),
+			},
+		},
+	}
+	jobName := "restore-job-" + cr.Name + "-" + cr.Spec.PXCCluster
+	volumeMounts := []corev1.VolumeMount{
+		{
+			Name:      "datadir",
+			MountPath: "/datadir",
+		},
+		{
+			Name:      "vault-keyring-secret",
+			MountPath: "/etc/mysql/vault-keyring-secret",
+		},
+	}
+	if pitr {
+		bucket := "operator-testing"
+		if cluster.Backup != nil && len(cluster.Backup.Storages) > 0 {
+			storage, ok := cluster.Backup.Storages[cr.Spec.PITR.BackupSource.StorageName]
+			if ok {
+				bucket = storage.S3.Bucket
+			}
+			fmt.Println("bucket from cr is", bucket)
+		}
+		command = []string{"pitr", "-action=r"}
+		envs = append(envs, corev1.EnvVar{
+			Name:  "RECOVERY_TYPE",
+			Value: cr.Spec.PITR.Type,
+		})
+		bcpNameArr := strings.Split(cr.Spec.PITR.BackupSource.Destination, "/")
+		bcpName := ""
+		if len(bcpNameArr) > 0 {
+			bcpName = bcpNameArr[len(bcpNameArr)-1]
+		}
+		envs = append(envs, corev1.EnvVar{
+			Name:  "BACKUP_NAME",
+			Value: bcpName,
+		})
+		envs = append(envs, corev1.EnvVar{
+			Name:  "S3_BUCKET",
```

Comment:
we already have `S3_BUCKET_URL` var for full backup here
```suggestion
			Name:  "PITR_S3_BUCKET_URL",
```